src/IECoreDelight/ParameterList.cpp

```diff
@@ -201,42 +201,50 @@ NSIParam_t ParameterList::parameter( const char *name, const IECore::Data *value
 		{
 			const vector<int> &v = static_cast<const IntVectorData *>( value )->readable();
 			result.type = NSITypeInteger;
+			result.arraylength = v.size();
+			result.flags |= NSIParamIsArray;
 			result.data = v.data();
-			result.count = v.size();
+			result.count = 1;
 			break;
 		}
 		case FloatVectorDataTypeId :
 		{
 			const vector<float> &v = static_cast<const FloatVectorData *>( value )->readable();
 			result.type = NSITypeFloat;
+			result.arraylength = v.size();
+			result.flags |= NSIParamIsArray;
 			result.data = v.data();
-			result.count = v.size();
+			result.count = 1;
 			break;
 		}
 		case V2fVectorDataTypeId :
 		{
 			const vector<V2f> &v = static_cast<const V2fVectorData *>( value )->readable();
 			result.type = NSITypeFloat;
-			result.arraylength = 2;
+			result.arraylength = v.size();
 			result.flags |= NSIParamIsArray;
 			result.data = v.data();
-			result.count = v.size();
+			result.count = 1;
 			break;
```

Comment:
From my search through the shader parameters, there are only single `V2f`-like parameters at the moment. Most of them are in the form of a UV coordinate, that really only make sense when getting an input from another shader.

But I think this may still be correct. It's a slight twist in wording, but the equivalent to 
> we're passing an array of floats with length equal to half the number of floats we actually have

is "We're passing twice as many floats as the number of items we're telling 3Delight we we have". That's the same as we're doing for colors or vectors - we pass 3 floats for every item we have, and 3Delight groups the triples as a single entry.

The main reason I'm interested in sorting out this particular case is that it can get rid of a warning we get when setting parameters for shaders that take UV coordinates :
`attempting to set parameter with wrong type: uvCoord (expected 'float[2]', received 'float')`

Reply:
I'm not sure I'm following. What I'm trying to say is that if `V2f` is `float[2]`, then `V2fVectorData` would be `float[2][size]`, but that's not expressible in the NSI API, unless `count` is used for the latter array length in this one case.

Oh, is this because there are input shader parameters with type `float[2]` and we're loading them as V2fVectorDataPlug, even though the shader author's intention was probably `V2fPlug`? But even in that case I don't think the new code is right. We'd have a V2fVectorData of size 1 (therefore 2 floats), and we're telling 3Delight that the array length is one (therefore one float).

If this is about the use of V2fVectorDataPlug, then ideally the shader would be changed to use `vector2` (and we'd update our loader to support that). In the absence of that, maybe we need to special-case the loading of `float[2]` to use V2fPlug?

Reply:
Right right, I was thinking it would get the size of the variable from the `float[2]` thing it serializes, like it gets implicitly for `color` and `vector`. But that's clear now that it's not the case. I added a warning for `V2fVectorData` that is being asked to be treated as an array in https://github.com/GafferHQ/gaffer/pull/5432/commits/2941b68c5362fde0c44212533ecc70d6614d9853 and reverted to the original behavior.